### PR TITLE
git: Add commit template

### PR DESCRIPTION
**Pull request by jlucovsky**
Issue: none

This commit adds a template that identifies commit message elements that we find important. The Suricata development team requests that contributions use the template to help improve commit messages. We reserve the right to strictly enforce adherence to the template in the future.

Configure git to use this template with:
    git config commit.template ..github/commit-template.txt

Describe changes:
- Add commit template with instructions on how to configure

### Provide values to any of the below to override the defaults.

- To use an LibHTP, Suricata-Verify or Suricata-Update pull request,
  link to the pull request in the respective `_BRANCH` variable.
- Leave unused overrides blank or remove.

SV_REPO=
SV_BRANCH=
SU_REPO=
SU_BRANCH=
LIBHTP_REPO=
LIBHTP_BRANCH=

### CODE REVIEW

File: .github/commit-template.txt [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Suricata commit template
+# See https://docs.suricata.io/en/latest/devguide/contributing/code-submission-process.html
+# Commit Title
+
+Issue: <Redmine issue number>
+
+# Description
+# Provide a description of the changes in this commit
+# - List individual changes
```
Comment (jufajardini) [inline .github/commit-template.txt:9]: maybe:
```suggestion
# - List main individual changes
```

**Review by jufajardini — COMMENTED** [review]
When we get this approved, it would be good to add a version of what is this PR description in the commit session about contributing: https://docs.suricata.io/en/latest/devguide/contributing/code-submission-process.html#commits

File: .github/commit-template.txt [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Suricata commit template
+# See https://docs.suricata.io/en/latest/devguide/contributing/code-submission-process.html
+# Commit Title
+
+Issue: <Redmine issue number>
```
Comment (jufajardini) [inline .github/commit-template.txt:5]: We should decide if we prefer to have the issue number in the beginning or end of the commit message. :P

Comment (jlucovsky) [reply .github/commit-template.txt]: Good point ... let's put it at the end of the message.

**Comment by victorjulien** [comment]
I like the idea, but I would expect the PR to contain a devguide update as well. It should explain how to use it.

**Comment by inashivb** [comment]
> I like the idea, but I would expect the PR to contain a devguide update as well. It should explain how to use it.

+1. I'd like to add that it should probably not live under `.github` as it is not a github specific thing.

**Review by catenacyber — CHANGES_REQUESTED** [review]
Please remove empty `sdfs` file ;-p

**Comment by jlucovsky** [comment]
Continued in #12212

**Pull request closed without merging**